tests/deadcode.py

```diff
@@ -0,0 +1,269 @@
+            return result
+
+        elif isinstance(node, ast.Constant):
+            return bool(node.value)
+        elif isinstance(node, ast.Name) and node.id == "__debug__":
+            return True
+
+        elif isinstance(node, ast.UnaryOp) and isinstance(node.op, ast.Not):
+            v = self.static_value(node.operand, deadcode)
+            if isinstance(v, bool):
+                return not v
+        else:
+            for n in ast.iter_child_nodes(node):
+                self.walk_deadcode(n, deadcode)
+
+        return None
+
+    def check_stmts(self, stmts, deadcode):
+        """
+        used to check the body: of a function, if, ...
+        """
+        for stmt in stmts:
+            stmt.deadcode = deadcode
+
+            if self.walk_deadcode(stmt, deadcode):
+                deadcode = True
+        return deadcode
+
+    def check_childs(self, childs, deadcode):
+        """
+        used to check childs: function arguments
+        """
+        for child in childs:
+            self.walk_deadcode(child, deadcode)
+
+    def walk_annotation(self, annotation, deadcode):
+        if self.future_annotations:
+            deadcode = True
+        self.walk_deadcode(annotation, deadcode)
+
+    def walk_deadcode(self, node, deadcode):
+        "returns True if this statement will never return"
+
+        # this check is not perfect but better than nothing
+        # it tries to prevent a lot of "node without associated Bytecode" errors
+
+        # They were generated test driven.
+        # Every case represented here is derived from a error where python performed dead code elimination.
+
+        if node is None:
+            return
+
+        if isinstance(node, list):
+            for child in node:
+                self.walk_deadcode(child, deadcode)
+            return
+
+        node.deadcode = deadcode or getattr(node, "deadcode", False)
+
+        if isinstance(node, ast.Module):
+            for stmt in node.body:
+                if isinstance(stmt, ast.ImportFrom):
+                    if stmt.module == "__future__" and any(
+                        "annotations" == alias.name for alias in stmt.names
+                    ):
+                        self.future_annotations = True
+
+            self.check_stmts(node.body, deadcode)
+        elif isinstance(node, (ast.With, ast.AsyncWith)):
+            self.check_childs(node.items, deadcode)
+            deadcode = self.check_stmts(node.body, deadcode)
+
+        elif isinstance(node, (ast.Return, ast.Break, ast.Continue, ast.Raise)):
+            if isinstance(node, ast.Raise):
+                self.walk_deadcode(node.exc, deadcode)
+                self.walk_deadcode(node.cause, deadcode)
+
+            if isinstance(node, ast.Return):
+                self.walk_deadcode(node.value, deadcode)
+
+            deadcode = True
+
+        elif isinstance(node, ast.Assert):
+            cnd = self.static_value(node.test, deadcode)
+
+            if cnd == False:
```

Comment:
```suggestion
            if cnd is False:
```

Please do this in general to make my linter happy

Reply:
which linter are you using? 
I saw also some comments to disable formatting. Are you also using any formatter? 
Can they be used with something like [pre-commit](https://pre-commit.com/)?

Reply:
Just PyCharm